### PR TITLE
fix(index): relax default in asset names `{RegExp}` (`options.test`)

### LINKED ISSUE #220: uglifyjs-webpack-plugin cannot uglify any file
this is my code:
```
new UglifyJsPlugin({
    uglifyOptions: {
         ie8: false,
	 ecma: 8,
	 //parse: {...options},
	 mangle: {
		 properties: {
		          // mangle property options
		  }
	  },
	 output: {
		   comments: false,
		   beautify: false
	  },
	 compress: {
	  },
	 warnings: false
    }
 })
```
 I used es7, es6, if i use webpack.optimize.UglifyJsPlugin , it can work normally.

```
"uglifyjs-webpack-plugin": "^1.1.8",   
 "webpack": "^3.10.0"
```

### PR DESCRIPTION
**Pull request by evilebottnawi**
### `Issues`

- Fixes #220

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ class UglifyJsPlugin {
 
     const {
       uglifyOptions = {},
-      test = /\.js$/i,
+      test = /.((js)?)(\?[a-z0-9]+)?$/i,
```
Comment (michael-ciniawsky) [inline src/index.js]: The `.` needs to be escaped (?)

```diff
- /.js
+ /\.js...
```

Why `((js)?)` ('nested' capturing groups) ?

**Comment by evilebottnawi** [comment]
@michael-ciniawsky done :+1:

**Comment by michael-ciniawsky** [comment]
Released in [`v1.2.3`](https://github.com/webpack-contrib/uglifyjs-webpack-plugin/releases/tag/v1.2.3) 🎉

**Comment by jazdw** [comment]
@evilebottnawi was it intentional to match the .j extension? Also an empty query should probably be allowed. Maybe `\.js(\?.*)?$`?

**Comment by michael-ciniawsky** [comment]
> was it intentional to match the .j extension?

The common **default** at this stage of the build is a JS asset ending in `.js` (`output.filename/output.chunkFilename`), it's possible to override this via `options.test` in case it's needed (but **optional**)

> Also an empty query should probably be allowed. Maybe \.js(\?.*)?$?

What's the 'use case' for an `empty` query? Why/When would someone do this ? But the `{RegExp}` could eventually be further relaxed/changed if there is a case made :)

**Comment by jazdw** [comment]
I'm not following you, maybe I didn't make myself clear. The Regex in the commit will match files with a .js and a .j extension which I don't think is desirable or the intention.

There's not a use case per say, just in case someone uses a variable substitution that evaluates to an empty string.

**Comment by jazdw** [comment]
@evilebottnawi @michael-ciniawsky any comment on the Regex?

**Comment by evilebottnawi** [comment]
@jazdw oh, yes, you are right we should fix it :+1:

**Pull request merged**